src/util/string.h

```diff
@@ -214,12 +215,9 @@ inline std::string MakeUnorderedList(const std::vector<std::string>& items)
 /**
  * Check if a string does not contain any embedded NUL (\0) characters
  */
-[[nodiscard]] inline bool ContainsNoNUL(std::string_view str) noexcept
+[[nodiscard]] inline bool ContainsNUL(std::string_view str) noexcept
```

Comment:
`[[nodiscard]]` doesn't make a lot of sense here, since the method doesn't have any side-effect

Reply:
I think I disagree here. If the function **had** side effects, it may make sense to not have [[nodiscard]] as you may intend to call the function, without caring what it returns. In this instance though, there is no purpose to call ContainsNUL and not use the returning value, so that would be indicative of a bug.

Reply:
what was the side effect and why is that important? Calling this without the return value would just be removed as dead code.